mathics/core/expression.py

```diff
@@ -743,9 +742,12 @@ def flatten(
         else:
             return self
 
-    def evaluate(self, evaluation) -> typing.Union["Expression", "Symbol"]:
-        from mathics.core.evaluation import ReturnInterrupt
-
+    # When we allow 3.9 only, the return type chould become type[Expression]
+    # See https://adamj.eu/tech/2021/05/16/python-type-hints-return-class-not-instance/
+    def evaluate(self, evaluation) -> typing.Type["Expression"]:
```

Comment:
OK, but ``evaluate`` can return a `Symbol`, or a `String` or a `Real` number among other things. All of them are instances of the class  ``BaseExpression``.

Reply:
Yes, I have this backwards. I mean `typing.Type["BaseExpression"]`. I will change that soon.

Reply:
BTW, this also suggests why the there might be a `LiteralExpression` class separate from `BaseExpression`.

In the case of `LiteralExpression` that is something that should not be subclassed, it is an end kind of thing of itself.

Reply:
BTW, in the reference, the example is:
```
def make_animal(animal_class: Type[Animal], name: str) -> Animal:
    return animal_class(name=name)
```
so `Type` seems to be used in arguments, but not in the return type. (weird, isn't it?)

Reply:
> BTW, this also suggests why the there might be a `LiteralExpression` class separate from `BaseExpression`.
> 
> In the case of `LiteralExpression` that is something that should not be subclassed, it is an end kind of thing of itself.

I am not fully sure to follow you. Maybe later a UML graph could be helpful.

Reply:
I think things will become clearer over time. Please be patient. 

Also, it is going to take a while to fully understand both this code and how WL and how that relates to conventional practice.

BTW, I have now updated the code and the docstring for Expression to be more clear about what is meant by "fixed point".  Please look over to make sure I have this correct.